### ISSUE TITLE
 Report redundant branch checks in GitHub Workflow jobs

### ISSUE THREAD

**Issue by ilya-hontarau** [rhysd/actionlint#434]
If a workflow is triggered only by pushes to a specific branch, any checks for this branch within the jobs are redundant and should be removed. For example:

```yaml
on:
  push:
    branches:
      - main

jobs:
  deploy:
    steps:
      - name: Deploy
        if: github.ref == 'refs/heads/main' # Unnecessary, this condition will always be true

```
Use Case:
During refactoring, it is easy to leave such unnecessary checks in the workflow.
